src/index.ts

```diff
@@ -577,165 +288,3 @@ export async function assemble<Annex extends AnnexParentType>(
-export async function gatherComponents(
-  setup: Partial<Omit<Components<FissionAccountsAnnex>, "account">> & Configuration
-): Promise<Components<FissionAccountsAnnex>>
-export async function gatherComponents(
-  setup: Partial<Components<any>> & Configuration
-): Promise<Components<any>> {
-  const config = extractConfig(setup)
-
-  const agent = setup.agent || await defaultComponents.agent(config)
-  const channel = setup.channel || await defaultComponents.channel()
-  const dns = setup.dns || await defaultComponents.dns()
-  const identifier = setup.identifier || await defaultComponents.identifier(config)
-  const manners = setup.manners || await defaultComponents.manners(config)
-  const storage = setup.storage || await defaultComponents.storage(config)
-
-  const depot = setup.depot || await defaultComponents.depot({ ...config, storage })
-  const account = setup.account || await defaultComponents.account({ agent, dns, manners })
-
-  return {
-    account,
-    agent,
-    channel,
-    depot,
-    dns,
-    identifier,
-    manners,
-    storage,
-  }
-}
-
-////////////////////////
-// DEFAULT COMPONENTS //
-////////////////////////
-
-export const defaultComponents = {
-  account(settings: {
-    agent: Agent.Implementation
-    dns: DNS.Implementation
-    manners: Manners.Implementation<FileSystem>
-  }): Promise<Account.Implementation<FissionAccountsAnnex>> {
-    return account.fission(settings)
-  },
-
-  async agent(config: Configuration): Promise<Agent.Implementation> {
-    const store = await storage.indexedDB({ name: `${namespace(config)}/agent` })
-    return agent.webCryptoAPI({ store })
-  },
-
-  channel(): Promise<Channel.Implementation> {
-    return channel.fission()
-  },
-
-  depot(settings: Configuration & { storage: Storage.Implementation }): Promise<Depot.Implementation> {
-    return depot.fissionBitswapIPFS(settings)
-  },
-
-  dns(): Promise<DNS.Implementation> {
-    return dns.doh()
-  },
-
-  async identifier(config: Configuration): Promise<Identifier.Implementation> {
-    const store = await storage.indexedDB({ name: `${namespace(config)}/identifier` })
-    return identifier.webCryptoAPI({ store })
-  },
-
-  manners(config: Configuration): Promise<Manners.Implementation<FileSystem>> {
-    return manners.default(config)
-  },
-
-  storage(config: Configuration): Promise<Storage.Implementation> {
-    return storage.indexedDB({ name: namespace(config) })
-  },
-}
-
-////////
-// 🛟 //
-////////
-
-/**
- * Is this browser supported?
- */
-export async function isSupported(): Promise<boolean> {
-  // Firefox in private mode can't use indexedDB properly,
-  // so we test if we can actually make a database.
-  try {
-    return await new Promise(resolve => {
-      const db = indexedDB.open("testDatabase")
-      db.onsuccess = () => resolve(true)
-      db.onerror = () => resolve(false)
-    })
-  } catch {
-    console.warn("IndexedDB is not supported")
-    return false
-  }
-}
```

Comment:
I moved this `isSupported` check to the storage component itself so this behaviour can be customised.